### PR TITLE
Fixed server and client NLA state machine for NULL identity.

### PR DESCRIPTION
**Pull request by akallabeth**
When using NULL credentials (current context)
the server state machine did not send back the
required authentication token.
On client side erroneous checks prevented sending
the appropriate public key.

@mfleisz Thx for the help ;)

### CODE REVIEW

**Comment by mfleisz** [comment]
+1

File: libfreerdp/core/gateway/ntlm.c [hunk 1]
```diff
@@ -54,19 +54,21 @@ BOOL ntlm_client_init(rdpNtlm* ntlm, BOOL http, char* user, char* domain, char*
...
 		return FALSE;
 	}
 
 	ntlm->cbMaxToken = ntlm->pPackageInfo->cbMaxToken;
 
 	status = ntlm->table->AcquireCredentialsHandle(NULL, NTLMSP_NAME,
-			SECPKG_CRED_OUTBOUND, NULL, &ntlm->identity, NULL, NULL,
-			&ntlm->credentials, &ntlm->expiration);
+						       SECPKG_CRED_OUTBOUND, NULL, &ntlm->identity, NULL, NULL,
+						       &ntlm->credentials, &ntlm->expiration);
```
Comment (bmiklautz) [inline libfreerdp/core/gateway/ntlm.c]: imho the formatting was better before..

File: libfreerdp/core/gateway/ntlm.c [hunk 2]
```diff
@@ -223,21 +225,33 @@ BOOL ntlm_authenticate(rdpNtlm* ntlm)
...
-			0, &ntlm->context, &ntlm->outputBufferDesc,
-			&ntlm->pfContextAttr, &ntlm->expiration);
-
+							(ntlm->haveContext) ? &ntlm->context : NULL,
+							(ntlm->ServicePrincipalName) ? ntlm->ServicePrincipalName : NULL,
+							ntlm->fContextReq, 0, SECURITY_NATIVE_DREP,
+							(ntlm->haveInputBuffer) ? &ntlm->inputBufferDesc : NULL,
+							0, &ntlm->context, &ntlm->outputBufferDesc,
+							&ntlm->pfContextAttr, &ntlm->expiration);
+
```
Comment (bmiklautz) [inline libfreerdp/core/gateway/ntlm.c]: I know there are no official guidelines on formatting for FreeRDP (expect read the different formatted code ;() but looking at other parts where indentation is used for long functions I'd say the extra intended part should start at the beginning of the function name (not on the brackets) or is intended with one (or two in extreme cases) extra tabs.

**Comment by MartinHaimberger** [comment]
+1 LGTM ... about the formating ... its a discussion for itself.

**Comment by bmiklautz** [comment]
@MartinHaimberger looking at the formatting of the other code it's not so much discussion imho..

**Pull request merged**